Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -18,276 +17,274 @@
-        event_list_for_push.append(event)
-        last_time = inc_time if inc_time else last_time
-        index += 1
-        if index == limit:
+    event_fields = [field['name'] for field in raw_response['fields']]
+    events_data = raw_response['data']
+
+    events = []
+    for index, event_data in enumerate(events_data):
+        if limit is not None and index == limit:  # No API-side limit param (needs to be managed on our end)
             break
-    return event_list_for_push, last_time, id_list
+
+        event: dict[str, Any] = {}
+        for field, value in zip(event_fields, event_data, strict=True):
+            event[field] = value if value != "-" else None  # "-" mark an empty field in Digital Guardian
+
+        add_time_and_profile_to_event(event, export_profile)
+
+        events.append(event)
+
+    return events
 
 
-def fetch_events(client: Client, last_run: dict[str, list], limit: int) -> Tuple[dict, list]:
+def get_events_command(client: Client, args: dict, export_profile: str) -> tuple[list[dict], dict, CommandResults]:
     """
+    Fetches a limited number of events and returns it in the CommandResults as a human readable markdown table.
     Args:
-        client (Client): DG client to use.
-        last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
-        limit (int):
-
+        client (Client): Digital Guardian client.
+        export_profile (str): Export profile name.
+        args (dict): Command arguments.
     Returns:
-        dict: Next run dictionary containing the timestamp that will be used in ``last_run`` on the next fetch and list of ID's
-              of ingested id's
-        list: List of events that will be created in XSIAM.
+        tuple[list, dict, CommandResults]: List of events, last run dictionary, CommandResults with human readable output.
     """
-    id_list = last_run.get('id_list', [])
-    last_time = str(last_run.get('start_time') or "")
-    demisto.debug('fetching events')
-    event_list = get_raw_events(client, last_time)
-    event_list_for_push, time_of_event, id_list = create_events_for_push(event_list, last_time, id_list, limit)
+    limit = arg_to_number(args.get('limit')) or 1000
+    events, last_run = fetch_events(client, export_profile, limit=limit)
+
+    human_readable = tableToMarkdown(name=f'Events for Profile {export_profile}', t=events, removeNull=True)
+    demisto.debug(f'Displayed limit of {limit} events from response')
 
-    # Save the next_run as a dict with the last_fetch key to be stored
-    next_run = {'start_time': time_of_event, 'id_list': id_list}
-    demisto.debug(f'Setting next run {next_run}.')
-    return next_run, event_list_for_push
+    return events, last_run, CommandResults(readable_output=human_readable)
 
 
-def add_time_to_events(events: list[dict]) -> None:
+def push_events(client: Client, events: list[dict], last_run: dict, export_profile: str, set_export_bookmark: bool) -> None:
```

Comment:
```suggestion
def push_events(client: Client, events: list[dict], last_run: dict, export_profile: str, *, set_export_bookmark: bool) -> None:
```
not required, but useful

Reply:
Done in commit 34998072bf2cd42577c899aa6f60313b6ad86751